controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,52 +224,74 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
+		// initialize in-memory structs
+		r.cephRadosNamespace = &rookCephv1.CephBlockPoolRadosNamespace{}
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
 		r.cephBlockPool.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephBlockPool" {
-				r.cephBlockPool.Name = res.Name
-				break
-			}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err := r.list(
+			cephRadosNamespaceList,
+			client.InNamespace(r.OperatorNamespace),
+			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
+		if err != nil {
+			return err
+		}
+
+		// if we found no CephBlockPoolRadosNamespaces, generate a new name
+		// if we found only one CephBlockPoolRadosNamespace with our query, we're good
+		// if we found more than one CephBlockPoolRadosNamespace, we can't determine which one to select, so error out
+		rnsItemsLen := len(cephRadosNamespaceList.Items)
+		if rnsItemsLen == 0 {
+			rnsNewName := fmt.Sprintf("cephradosnamespace-%s-%s", r.StorageClassRequest.Spec.StorageProfile, r.storageConsumer.Name)
+			r.log.V(1).Info("no valid CephBlockPoolRadosNamespace found, creating new one", "CephBlockPoolRadosNamespace", rnsNewName)
+			r.cephRadosNamespace.Name = rnsNewName
+		} else if rnsItemsLen == 1 {
+			cephRns := cephRadosNamespaceList.Items[0]
+			r.cephBlockPool.Name = cephRns.Spec.BlockPoolName
+			r.cephRadosNamespace = &cephRns
+			r.log.V(1).Info("valid CephBlockPoolRadosNamespace found", "CephBlockPoolRadosNamespace", r.cephRadosNamespace.Name)
+		} else {
+			return fmt.Errorf("invalid number of CephBlockPoolRadosNamespaces for storage consumer %q and storage profile %q: found %d, expecting 0 or 1", r.storageConsumer.Name, profileName, rnsItemsLen)
 		}
 
-		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		// check if a cephblockpool resource exists for the desired storageprofile.
 		if r.cephBlockPool.Name == "" {
```

Comment:
How can we get here without a name? 
If the request has a storage profile defined then we have a name. 
If not we already have the code above to set the profile name as the default block pool name. 

This means we always have a name and the code under the if will never execute.

Reply:
Holdover from deleted code. Fixed.